ui/apps/ui/src/app/components/sort-by-functionality/sort-by-functionality.component.ts

```diff
@@ -0,0 +1,146 @@
+import { SortOptionValuesEnum } from '@components/sort-by-functionality/sort-option-values.enum';
+
+@UntilDestroy()
+@Component({
+  selector: 'ess-sort-by-functionality',
+  template: `<div class="col-sm-4 sort_container">
+    <label for="sorts">Sort By</label>
+    <select
+      [(ngModel)]="selectedSortOption"
+      #sorts
+      id="sorts"
+      label="Sort by"
+      class="form-select"
+      (change)="this.onSelected(sorts.value)"
+    >
+      <option *ngFor="let sortBy of sortOptions" value="{{ sortBy.value }}">
+        {{ sortBy.description }}
+      </option>
+    </select>
+  </div>`,
+  styles: [
+    `
+      :host {
+        display: contents;
+      }
+
+      label {
+        width: 100px;
+        font-family: 'Inter';
+        font-style: normal;
+        font-weight: 400;
+        font-size: 14px;
+        line-height: 20px;
+        color: #4f4f4f;
+      }
+
+      select {
+        font-family: 'Inter';
+        font-style: normal;
+        font-weight: 500;
+        font-size: 14px;
+        line-height: 24px;
+        color: #101828;
+      }
+
+      select:focus {
+        -webkit-box-shadow: none;
+        box-shadow: none;
+        border: 1px solid #ced4da;
+      }
+
+      .sort_container {
+        width: 18rem;
+        display: flex;
+        align-items: center;
+      }
+    `,
+  ],
+})
+export class SortByFunctionalityComponent implements OnInit {
+  public selectedSortOption: string = DefaultSortValueEnum.DEFAULT;
+
+  public readonly sortOptions: Array<ISortOption> = [
+    {
+      description: SortOptionValuesEnum.RELEVANCY,
+      value: DefaultSortValueEnum.RELEVANCY,
+    },
+    {
+      description: SortOptionValuesEnum.DATE_MOST_RECENT,
+      value: DefaultSortValueEnum.DATE_MOST_RECENT,
+    },
+    {
+      description: SortOptionValuesEnum.DATE_LEAST_RECENT,
+      value: DefaultSortValueEnum.DATE_LEAST_RECENT,
+    },
+    {
+      description: SortOptionValuesEnum.MOST_POPULAR,
+      value: DefaultSortValueEnum.MOST_POPULAR,
+    },
+  ];
+
+  private searchCollections = this._navConfigsRepository.getAll();
+  private collectionFc = new FormControl<ICollectionNavConfig>(
+    this.searchCollections[0],
+    { nonNullable: true }
+  );
+
+  constructor(
+    private _customRoute: CustomRoute,
+    private _navConfigsRepository: NavConfigsRepository,
+    private _router: Router
+  ) {}
+
+  ngOnInit() {
+    this._customRoute.s$
+      .pipe(
+        untilDestroyed(this),
+        filter((s: string) => s !== DefaultSortValueEnum.DEFAULT)
+      )
+      .subscribe((s) => {
+        this.selectedSortOption = s;
+      });
+  }
+
+  async updateQueryParams(s: string, $event: Event | null = null) {
+    let searchQueryParamsQ;
+
+    this._customRoute.q$
+      .pipe(
+        untilDestroyed(this),
+        filter((q: string) => q !== '*')
+      )
+      .subscribe((q) => {
+        searchQueryParamsQ = q;
+      });
```

Comment:
This subscription will cause memory leak and will be spawned for every call of the updateQueryParams, which is probably not something we want.
Also spawning two async functions (first this and then navigate) is probably not the best idea. I would either chain it with `.navigate` or wrap it in await.

Reply:
I solved differently